### PR TITLE
Support ws or wss proto

### PR DESCRIPTION
**Pull request by jakemcdermott**
##### SUMMARY
Use `ws` when on `http` and `wss` when on `https`

related: https://github.com/ansible/awx/issues/8881

### CODE REVIEW

File: awx/ui_next/src/setupTests.js [hunk 1]
```diff
@@ -24,3 +24,10 @@ global.console = {
 // and so this mock ensures that we don't encounter a reference error
 // when running the tests
 global.__webpack_nonce__ = null;
+
+// This is to make websocket-using components behave as if they're connecting via
+// wss, which is the default behavior being tested in most websocket tests
+Object.defineProperty(window, 'location', {
+  value: new URL('https://localhost'),
+});
+window.location.replace = jest.fn();
```
Comment (jakemcdermott) [inline awx/ui_next/src/setupTests.js]: This may be a bit heavy-handed, but the mock seems to work fine for the tests we have so far. :man_shrugging:

**Comment by unlikelyzero** [comment]
Note: no additional coverage is needed. Once e2e is 🟢 this will be implicitly tested

**Review by keithjgrant — APPROVED** [review]

**Review by AlexSCorey — APPROVED** [review]
This looks good to me.

**Comment by jakemcdermott** [comment]
re:

> Note: no additional coverage is needed. Once e2e is green_circle this will be implicitly tested

1 e2e failure: https://dashboard.cypress.io/projects/ui2mq7/runs/3606/overview

...but it's the workflow visualizer notifications tab test which I've observed to be flakey and failing occasionally on other branches: e.g: https://dashboard.cypress.io/projects/ui2mq7/runs/3607/overview

**Comment by jakemcdermott** [comment]
related: https://github.com/ansible/awx/issues/8932

**Pull request merged**